renin/src/ui/AudioBar.ts

```diff
@@ -86,31 +86,37 @@ export class AudioBar {
     return lerp(this.zoomStartFrame, this.zoomEndFrame, xInPercent);
   }
 
-  zoom(delta: number) {
+  zoom(delta: number, center: number = 0) {
     if (!this.music) {
       return;
     }
-    this.zoomAmount = Math.max(1, this.zoomAmount * delta);
+    this.zoomAmount = clamp(1, this.zoomAmount * delta, 100);
     const maxFrame = (this.music.getDuration() * 60) | 0;
     if (this.zoomAmount === 1) {
       this.zoomStartFrame = 0;
       this.zoomEndFrame = maxFrame;
       return;
     }
-    const currentFrame = this.renin.frame;
-    const currentFramePercentage = (currentFrame - this.zoomStartFrame) / (this.zoomEndFrame - this.zoomStartFrame);
     const newFrameWidth = maxFrame / this.zoomAmount;
-    this.zoomStartFrame = currentFrame - currentFramePercentage * newFrameWidth;
-    this.zoomEndFrame = currentFrame + (1 - currentFramePercentage) * newFrameWidth;
+    const centerFrame = this.zoomStartFrame + newFrameWidth * center;
```

Comment:
I think maybe this should have been oldFrameWidth.

Reply:
I think so too.